Core/src/org/sleuthkit/autopsy/geolocation/MapPanel.java

```diff
@@ -685,17 +689,85 @@ private void zoomOutBtnActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FI
      * Renderer for the map waypoints.
      */
     private class MapWaypointRenderer implements WaypointRenderer<MapWaypoint> {
+        private final Map<Color, BufferedImage> imageCache = new HashMap<>();
```

Comment:
This class is probably big enough for its own class file now.  Leaving it in MapPanel just bloats the file.

Reply:
Okay, I'll move it out.

Reply:
MapWaypoint is not public, so it would require making it so to move this class out from being an inner class. I don't really think it's that large. Thoughts @rcordovano ?

Reply:
You would not have to make it public.  By specifying neither public nor private at the class definition the class will be visible to all other classes in the package.  

Please move it to its own class file.  I don't recall at the moment why I made MapPanel public that might be erroneous.

Reply:
@ethteck and @kellykelly3, some thoughts...

None of this code is public API, i.e., none of the packages we are discussing are visible to code outside of this NetBeans module / JAR. That give us some welcome flexibility in refactoring and evolving the code over time, versus getting it absolutely right the first time. This is a good thing.

The MapWaypoint class does seem to be fairly large, so that is an argument in favor of giving it its own file. However, that is not the only criterion for judging whether a class should be an inner class, at least in my opinion. In fact, I have I have wrestled long and hard with these types of decisions, and have not come to a universal conclusion or reliable rule of thumb. I think as a general guideline, if something has no meaning outside of its use in another class, it is a good candidate for an inner class. From what I am seeing, it looks like MapWaypoint is a view class wrapped around a Waypoint model class with usage outside of a MapPanel (but please correct me if I am wrong, I am guilty of code skimming here), so it does seem to merit its own file. It looks that has already been done, so I think that's fine. We can change it later if we wish.

Reply:
@rcordovano I may have misunderstood your answer, but it seems like you're talking about whether MapWaypoint should have its own file. However, what Kelly is suggesting is that MapWaypointRenderer be its own file. Do you have an opinion on this? 

I tried to do so earlier, and for some reason I was getting errors saying MapWaypoint wasn't public despite both being classes being in the same package. Maybe that was just the IDE being buggy. I can try again if you think that's what we should do.

Reply:
@ethteck and @kellykelly3, I _was_ talking about the wrong class, which explains a lot. I apologize for any confusion caused by my confusion. I have to go with Ethan's argument for the location of MapWaypointRenderer. It is a small, private class which has no use outside of MapPanel. If we want to cut down the size of the MapPanel class, I think we need to look elsewhere.